docs/concepts/workloads/containers/container-lifecycle-hooks.md

```diff
@@ -0,0 +1,163 @@
+as well as the hooks and lifecycle that allow Containers to interact with the management system.
+{% endcapture %}
+
+{:toc}
+
+{% capture body %}
+The Kubernetes Container environment defines a series of Hooks that are available to Containers that register the corresponding Hook handlers.
+Container Hooks are analogous to operating system signals in a traditional process model;
+however, their purpose is to make it easier to build reliable and scalable cloud applications in a Kubernetes cluster. 
+
+Containers that participate in a cluster lifecycle are considered *cluster native*.
+
+
+## Container environment
+
+The Container environment provides several important resources to a Container.
+This includes the filesystem that is available to the container, 
+which is a combination of an [image](/docs/concepts/workloads/containers/images) and one or more [volumes](/docs/concepts/storage/volumes).
+
+There are also two types of information that are available within the Container environment:
+
+* Information about the container itself.
+* Information about other objects in the system.
+
+### Container information
+
+The *hostname* of the container is the name of the pod in which the container is running. 
+It is available through the `hostname` command or the [`gethostname`](http://man7.org/linux/man-pages/man2/gethostname.2.html) function call in libc.
+
+The pod name and namespace are also available as environment variables through the [downward API](docs/tasks/configure-pod-container/downward-api-volume-expose-pod-information). 
+Additionally, user defined environment variables from the pod definition are also available to the container, 
+as are any environment variables specified statically in the Docker image.
+
+### Cluster information
+
+A list of all services that were running when a container was created via the Kubernetes Cluster API are available to that container as environment variables.
+The set of environment variables matches the syntax of Docker links.
+
+For a service named *foo* that maps to a container port named *bar*, 
+the following variables are defined:
+
+```shell
+FOO_SERVICE_HOST=<the host the service is running on>
+FOO_SERVICE_PORT=<the port the service is running on>
+```
+
+Services have dedicated IP addresses and are available to the container via DNS
+if [DNS addon](http://releases.k8s.io/{{page.githubbranch}}/cluster/addons/dns/) is enabled. 
+Because DNS is still not an enumerable protocol, 
+environment variables are provided so that Containers can do discovery.
+
+## Container hooks
+
+Container hooks provide information to the container about events in its management lifecycle.  
+For example, immediately after a container is started, it receives a `PostStart` hook.  
+Hooks are broadcast in the container with information about the lifecycle of the container.  
+They are different from the events provided by Docker or other systems which are output by the container.  
```

Comment:
I think these sentences about Docker and output events take the reader on an unnecessary side trip. I would remove them.